packages/flame/lib/src/game/mixins/game.dart

```diff
@@ -42,8 +42,32 @@ mixin Game {
   Vector2? _size;
 
   /// This variable ensures that Game's [onLoad] is called no more than once.
+  late final Future<void>? _onLoadFuture = onLoad();
+
+  bool _debugOnLoadStarted = false;
+
   @internal
-  late Future<void>? onLoadFuture = onLoad();
+  Future<void>? get onLoadFuture {
+    assert(
+      () {
+        _debugOnLoadStarted = true;
+        return true;
+      }(),
+    );
+    return _onLoadFuture;
+  }
+
+  /// To be used for tests that needs to evaluate the game after it has been
+  /// loaded by the game widget.
+  @visibleForTesting
+  Future<void>? toBeLoaded() {
```

Comment:
That's conceptually the same future as `Component.loaded`, isn't it? Why not name it the same?